warehouse/accounts/security_policy.py

```diff
@@ -95,11 +98,17 @@ def _basic_auth_check(username, password, request):
                 )
 
             login_service.update_user(user.id, last_login=datetime.datetime.utcnow())
-            user.record_event(
+            has_seen_ip_before = login_service.seen_from_ip_before(
```

Comment:
I'm worried doing this for BasicAuth (aka uploads) will be way too noisy. Consider a user that only uploads via GitHub Actions, they may get a new notice every time a release workflow runs.

Reply:
Correct, this should be the only place where HTTP Basic Auth is allowed and is used solely for the uploads endpoint. 2FA users will not be able to use the BasicAuth method due to the `_check_mfa` that is called by the permits part, they will have to use a macaroon directly or via trusted publisher.